src/Symfony/Component/DependencyInjection/Tests/EnvVarProcessorTest.php

```diff
@@ -0,0 +1,304 @@
+            $this->assertSame('foo', $name);
+
+            return $value;
+        });
+
+        $this->assertSame($processed, $result);
+    }
+
+    public function validConsts()
+    {
+        return array(
+            array('Symfony\Component\DependencyInjection\Tests\EnvVarProcessorTest::TEST_CONST', self::TEST_CONST),
+            array('E_ERROR', E_ERROR),
+        );
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage undefined constant
+     * @dataProvider invalidConsts
+     */
+    public function testGetEnvConstInvalid($value)
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $processor->getEnv('const', 'foo', function ($name) use ($value) {
+            $this->assertSame('foo', $name);
+
+            return $value;
+        });
+    }
+
+    public function invalidConsts()
+    {
+        return array(
+            array('Symfony\Component\DependencyInjection\Tests\EnvVarProcessorTest::UNDEFINED_CONST'),
+            array('UNDEFINED_CONST'),
+        );
+    }
+
+    public function testGetEnvBase64()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $result = $processor->getEnv('base64', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return base64_encode('hello');
+        });
+
+        $this->assertSame('hello', $result);
+    }
+
+    public function testGetEnvJson()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $result = $processor->getEnv('json', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return json_encode(array(1));
+        });
+
+        $this->assertSame(array(1), $result);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage Syntax error
+     */
+    public function testGetEnvInvalidJson()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $processor->getEnv('json', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return 'invalid_json';
+        });
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage Invalid JSON env var
+     * @dataProvider otherJsonValues
+     */
+    public function testGetEnvJsonOther($value)
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $processor->getEnv('json', 'foo', function ($name) use ($value) {
+            $this->assertSame('foo', $name);
+
+            return json_encode($value);
+        });
+    }
+
+    public function otherJsonValues()
+    {
+        return array(
+            array(1),
+            array(1.1),
+            array(true),
+            array(false),
```

Comment:
I left out `null` as it would conflict with my other PR, https://github.com/symfony/symfony/pull/26498

Reply:
Or should I better include it here and update in second PR assuming this gets merged first?